app/src/main/java/ch/epfl/qedit/backend/Backend.java

```diff
@@ -0,0 +1,13 @@
+package ch.epfl.qedit.backend;
+
+import ch.epfl.qedit.frontendBackendInterface.FallibleEntity;
+import ch.epfl.qedit.frontendBackendInterface.Request;
+import ch.epfl.qedit.frontendBackendInterface.Response;
+
+public abstract class Backend extends FallibleEntity {
```

Comment:
We could consider making it an interface maybe ?

Reply:
Why ? Do you want to create dozens of backends ? I don't even understand why he created an abstract class for the backend

Reply:
Because we want to make the application independent of the backend running behind it. We abstract it away. This makes switching the backend easy. It's not our goal to "create dozens of backends".

Reply:
We can create a backend class and then hide all the different functions of our server, namely Firebase in it. Something like an adapter, which I did it in a `Server class`. But you it is not what you seem to want to do. You don't even hide firebase later because you created a class in the open for all our application to see.

Reply:
Well the code for the real backend has to be somewhere ... By hiding I mean that we only use the abstractions in the apps code except for the code that implements the real backend. I don't see how you want to circumvent having that code somewhere. And I if you'd open a PR we could actually see what your doing.

Reply:
Oh yeah. I see now.